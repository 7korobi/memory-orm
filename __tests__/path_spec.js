/*
 * decaffeinate suggestions:
 * DS102: Remove unnecessary code created because of implicit returns
 * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
 */
require('./models/index')
const Mem = require('../lib/index')

describe('path', function () {
  test('children snapshot', () => {
    expect(Mem.Query.paths.pluck('id', 'paths.list')).toMatchSnapshot()
  })
  test('parent snapshot', () => {
    expect(Mem.Query.paths.pluck('id', 'path')).toMatchSnapshot()
  })
  test('reduce id snapshot', () => {
    expect(Mem.Query.paths.reduce.id).toMatchSnapshot()
  })
})
