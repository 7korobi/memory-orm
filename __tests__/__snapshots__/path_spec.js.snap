// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`path children snapshot 1`] = `
list [
  Array [
    "book-1-aaa",
    list [],
  ],
  Array [
    "book-1-bbb",
    list [
      model {
        "_id": "book-1-bbb-ccc",
        "book_id": "book-1",
        "folder_id": "book",
        "idx": "ccc",
        "path_id": "book-1-bbb",
      },
      model {
        "_id": "book-1-bbb-ddd",
        "book_id": "book-1",
        "folder_id": "book",
        "idx": "ddd",
        "path_id": "book-1-bbb",
      },
    ],
  ],
  Array [
    "book-1-bbb-ccc",
    list [],
  ],
  Array [
    "book-1-bbb-ddd",
    list [
      model {
        "_id": "book-1-bbb-ddd-eee",
        "book_id": "book-1",
        "folder_id": "book",
        "idx": "eee",
        "path_id": "book-1-bbb-ddd",
      },
    ],
  ],
  Array [
    "book-1-bbb-ddd-eee",
    list [],
  ],
  Array [
    "book-1-ccc-ddd",
    list [],
  ],
  Array [
    "book-1-ccc-eee",
    list [],
  ],
  Array [
    "book-1-ddd-eee",
    list [],
  ],
  Array [
    "book-1-fff-ggg-hhh-iii",
    list [],
  ],
]
`;

exports[`path parent snapshot 1`] = `
list [
  Array [
    "book-1-aaa",
    null,
  ],
  Array [
    "book-1-bbb",
    null,
  ],
  Array [
    "book-1-bbb-ccc",
    model {
      "_id": "book-1-bbb",
      "book_id": "book-1",
      "folder_id": "book",
      "idx": "bbb",
    },
  ],
  Array [
    "book-1-bbb-ddd",
    model {
      "_id": "book-1-bbb",
      "book_id": "book-1",
      "folder_id": "book",
      "idx": "bbb",
    },
  ],
  Array [
    "book-1-bbb-ddd-eee",
    model {
      "_id": "book-1-bbb-ddd",
      "book_id": "book-1",
      "folder_id": "book",
      "idx": "ddd",
      "path_id": "book-1-bbb",
    },
  ],
  Array [
    "book-1-ccc-ddd",
    null,
  ],
  Array [
    "book-1-ccc-eee",
    null,
  ],
  Array [
    "book-1-ddd-eee",
    null,
  ],
  Array [
    "book-1-fff-ggg-hhh-iii",
    null,
  ],
]
`;

exports[`path reduce id snapshot 1`] = `
Object {
  "navi": Object {
    "book-1": Object {
      "book-1-aaa": 0,
      "book-1-bbb": Object {
        "book-1-bbb-ccc": 0,
        "book-1-bbb-ddd-eee": 0,
      },
      "book-1-ccc": Object {
        "book-1-ccc-ddd": 0,
        "book-1-ccc-eee": 0,
      },
      "book-1-ddd-eee": 0,
      "book-1-fff-ggg-hhh-iii": 0,
    },
  },
}
`;
